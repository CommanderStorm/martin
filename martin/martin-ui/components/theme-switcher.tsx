"use client";

import { Moon, Sun, SunMoon } from "lucide-react";
import { useTheme } from "next-themes";
import * as React from "react";
import { Button } from "./ui/button";
import {
  Tooltip,
  TooltipContent,
  TooltipTrigger,
} from "./ui/tooltip";

const themeOrder = ["light", "dark", "system"] as const;

export function ThemeSwitcher() {
	const { theme, setTheme } = useTheme();

	const getNextTheme = () => {
		const idx = themeOrder.indexOf((theme as typeof themeOrder[number]) || "system");
		return themeOrder[(idx + 1) % themeOrder.length];
	};

	const getIcon = () => {
		switch (theme) {
			case "light":
				return <Sun className="h-[1.2rem] w-[1.2rem]" aria-hidden="true" />;
			case "dark":
				return <Moon className="h-[1.2rem] w-[1.2rem]" aria-hidden="true" />;
			default:
				return <SunMoon className="h-[1.2rem] w-[1.2rem]" aria-hidden="true" />;
		}
	};

	const getLabel = () => {
		switch (theme) {
			case "light":
				return "Switch to dark theme";
			case "dark":
				return "Switch to system theme";
			default:
				return "Switch to light theme";
		}
	};

	return (
		<Tooltip>
			<TooltipTrigger asChild>
				<Button
					variant="outline"
					size="icon"
					onClick={() => setTheme(getNextTheme())}
					aria-label={getLabel()}
				>
					{getIcon()}
				</Button>
			</TooltipTrigger>
			<TooltipContent>
				<p>{getLabel()}</p>
			</TooltipContent>
		</Tooltip>
	);
}
