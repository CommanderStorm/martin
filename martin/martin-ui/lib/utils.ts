import { type ClassValue, clsx } from "clsx";
import { twMerge } from "tailwind-merge";

export function cn(...inputs: ClassValue[]) {
	return twMerge(clsx(inputs));
}

/**
 * Formats a file size in bytes to a human-readable string
 * @param bytes The file size in bytes
 * @returns A formatted string representation of the file size
 */
export function formatFileSize(bytes: number): string {
	if (bytes === 0) return "0 Bytes";
	if (!bytes || isNaN(bytes) || bytes < 0) return "Unknown size";

	const k = 1024;
	const sizes = ["Bytes", "KB", "MB", "GB", "TB"];
	const i = Math.floor(Math.log(bytes) / Math.log(k));

	// Handle potential overflow or incorrect values
	if (i >= sizes.length) return "File too large";

	return parseFloat((bytes / k ** i).toFixed(2)) + " " + sizes[i];
}
