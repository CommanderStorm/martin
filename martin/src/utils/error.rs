use std::error::Error;
use std::fmt::Write as _;
use std::io;
use std::path::PathBuf;

/// A convenience [`Result`] for Martin crate.
pub type MartinResult<T> = Result<T, MartinError>;

fn elide_vec(vec: &[String], max_items: usize, max_len: usize) -> String {
    let mut s = String::new();
    for (i, v) in vec.iter().enumerate() {
        if i > max_items {
            let _ = write!(s, " and {} more", vec.len() - i);
            break;
        }
        if i > 0 {
            s.push(' ');
        }
        if v.len() > max_len {
            s.push_str(&v[..max_len]);
            s.push('â€¦');
        } else {
            s.push_str(v);
        }
    }
    s
}

#[derive(thiserror::Error, Debug)]
pub enum MartinError {
    #[error("The --config and the connection parameters cannot be used together. Please remove unsupported parameters '{}'", elide_vec(.0, 3, 15))]
    ConfigAndConnectionsError(Vec<String>),

    #[error("Unable to bind to {1}: {0}")]
    BindingError(io::Error, String),

    #[error("Unrecognizable connection strings: {0:?}")]
    UnrecognizableConnections(Vec<String>),

    #[cfg(any(
        feature = "postgres",
        feature = "pmtiles",
        feature = "mbtiles",
        feature = "cog"
    ))]
    #[error(transparent)]
    TileSourceError(#[from] martin_core::tiles::TileSourceError),

    #[error(transparent)]
    ConfigFileError(#[from] crate::config::file::ConfigFileError),

    #[cfg(feature = "sprites")]
    #[error(transparent)]
    SpriteError(#[from] martin_core::sprites::SpriteError),

    #[cfg(feature = "fonts")]
    #[error(transparent)]
    FontError(#[from] martin_core::fonts::FontError),

    #[error(transparent)]
    WebError(#[from] actix_web::Error),

    #[error(transparent)]
    IoError(#[from] io::Error),

    #[error("Internal error: {0}")]
    InternalError(#[from] Box<dyn Error + Send + Sync>),
}
