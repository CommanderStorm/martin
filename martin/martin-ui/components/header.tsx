"use client";

import { BookOpen, Info } from "lucide-react";
import Image from "next/image";
import Link from "next/link";
import { useTheme } from "next-themes";
import { Badge } from "@/components/ui/badge";
import { HoverCard, HoverCardContent, HoverCardTrigger } from "@/components/ui/hover-card";
import { ThemeSwitcher } from "./theme-switcher";
import { Skeleton } from "./ui/skeleton";

export function Header() {
  const { theme } = useTheme();
  return (
    <header className="border-b bg-navbar backdrop-blur-md backdrop-brightness-90 sticky top-0 z-50">
      <div className="container mx-auto px-6">
        <div className="flex h-20 items-center justify-between">
          <div className="flex items-center space-x-4">
            <Link className="items-center flex" href="/">
              {theme ?
              <Image
                alt="Martin Logo"
                className="-mt-5 -rotate-6"
                height={48}
                priority
                src={`/logo_martin-${theme}.svg`}
                title="Martin Tile Server"
                width={96}
              />:<Skeleton className="h-12 w-24" />}
              <h1 className="text-3xl -ms-2 font-bold leading-relaxed text-foreground select-none">
                MARTIN
              </h1>
            </Link>
            <Badge asChild className="hover:bg-purple-700 hidden md:block" variant="default">
              <Link
                className="p-1"
                href={`https://github.com/maplibre/martin/releases/tag/${process.env.NEXT_PUBLIC_VERSION}`}
              >
                {process.env.NEXT_PUBLIC_VERSION}
              </Link>
            </Badge>
          </div>
          <div className="flex items-center space-x-6">
            <HoverCard>
              <HoverCardTrigger asChild>
                <Link
                  className="flex items-center gap-2 text-foreground hover:bg-accent hover:text-accent-foreground px-3 py-2 rounded-md transition-all"
                  href="https://maplibre.org/martin/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <BookOpen size={18} />
                  <span>Documentation</span>
                </Link>
              </HoverCardTrigger>
              <HoverCardContent className="w-80">
                <div className="space-y-2">
                  <h4 className="text-sm font-semibold">Martin Documentation</h4>
                  <p className="text-sm">
                    Access comprehensive guides and documentation for the Martin tile server.
                  </p>
                </div>
              </HoverCardContent>
            </HoverCard>
            <HoverCard>
              <HoverCardTrigger asChild>
                <Link
                  className="md:flex hidden items-center gap-2 text-foreground hover:bg-accent hover:text-accent-foreground px-3 py-2 rounded-md transition-all"
                  href="https://maplibre.org"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <Info size={18} />
                  <span>About us</span>
                </Link>
              </HoverCardTrigger>
              <HoverCardContent className="w-80">
                <div className="space-y-2">
                  <h4 className="text-sm font-semibold">About MapLibre</h4>
                  <p className="text-sm">
                    Learn about MapLibre,
                    <br />
                    the open-source collective behind this project.
                  </p>
                </div>
              </HoverCardContent>
            </HoverCard>
            <ThemeSwitcher />
          </div>
        </div>
      </div>
    </header>
  );
}
