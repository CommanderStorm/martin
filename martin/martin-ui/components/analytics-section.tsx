import { Image, Layers, Palette, Type } from "lucide-react";
import { ErrorState } from "@/components/error/error-state";
import { Card, CardContent, CardHeader, CardTitle } from "@/components/ui/card";
import { Skeleton } from "@/components/ui/skeleton";
import type { AnalyticsData } from "@/lib/types";

interface AnalyticsSectionProps {
  analytics?: AnalyticsData;
  isLoading?: boolean;
  error?: Error | null;
  onRetry?: () => void;
  isRetrying?: boolean;
}

const CARD_CONFIG = [
  {
    icon: Layers,
    key: "tiles",
    title: "Tiles",
  },
  {
    icon: Palette,
    key: "styles",
    title: "Styles",
  },
  {
    icon: Type,
    key: "fonts",
    title: "Fonts",
  },
  {
    icon: Image,
    key: "sprites",
    title: "Sprites",
  },
] as const;

export function AnalyticsSection({
  analytics,
  isLoading,
  error = null,
  onRetry,
  isRetrying = false,
}: AnalyticsSectionProps) {
  if (error) {
    return (
      <div className="mb-8">
        <ErrorState
          description="Unable to fetch server metrics and usage data"
          error={error}
          isRetrying={isRetrying}
          onRetry={onRetry}
          showDetails={true}
          title="Failed to Load Analytics"
          variant="server"
        />
      </div>
    );
  }

  return (
    <div className="space-y-6 mb-8">
      <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-4 gap-6">
        {CARD_CONFIG.map(({ key, title, icon: Icon }) => {
          const data = analytics?.[key as keyof AnalyticsData];
          return (
            <Card key={key}>
              <CardHeader className="flex flex-row items-center justify-between space-y-0 pb-2">
                <CardTitle className="text-sm font-medium">{title}</CardTitle>
                <Icon className="h-4 w-4 text-primary" />
              </CardHeader>
              <CardContent>
                <div className="text-2xl font-bold flex flex-row gap-1">
                  {isLoading || !data ? (
                    <Skeleton className="h-6 w-8 flex flex-row" />
                  ) : (
                    Math.round(data.averageRequestDurationMs)
                  )}{" ms"}
                </div>
                <p className="text-xs text-muted-foreground flex flex-row gap-1">
                  {isLoading || !data ? (
                    <Skeleton className="h-3 w-12" />
                  ) : (
                    data.requestCount.toLocaleString()
                  )}{" requests"}
                </p>
              </CardContent>
            </Card>
          );
        })}
      </div>
    </div>
  );
}
