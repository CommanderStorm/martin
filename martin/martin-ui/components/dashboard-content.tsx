"use client";

import { useRouter, useSearchParams } from "next/navigation";
import { type ErrorInfo, useCallback, useEffect } from "react";
import { FontCatalog } from "@/components/catalogs/font";
import { SpriteCatalog } from "@/components/catalogs/sprite";
import { StylesCatalog } from "@/components/catalogs/styles";
import { TilesCatalog } from "@/components/catalogs/tiles";
import { ErrorBoundary } from "@/components/error/error-boundary";
import { Tabs, TabsContent, TabsList, TabsTrigger } from "@/components/ui/tabs";
import { Toaster } from "@/components/ui/toaster";
import { useAsyncOperation } from "@/hooks/use-async-operation";
import { useToast } from "@/hooks/use-toast";
import { buildMartinUrl } from "@/lib/api";
import { getMartinMockCatalog } from "@/lib/mockResponses";
import type { CatalogSchema } from "@/lib/types";

const fetchCatalog = async (): Promise<CatalogSchema> => {
  if (process.env.NEXT_PUBLIC_MARTIN_ENABLE_MOCK_API === "true") {
    return getMartinMockCatalog();
  }

  const res = await fetch(buildMartinUrl("/catalog"));
  if (!res.ok) {
    throw new Error(`Failed to fetch catalog: ${res.statusText}`);
  }
  return res.json();
};

export function DashboardContent() {
  const { toast } = useToast();
  const router = useRouter();
  const searchParams = useSearchParams();

  const searchQuery = searchParams.get("search") || "";
  // Update search param in URL, preserving tab and other params
  const setSearchQuery = (query: string) => {
    const params = new URLSearchParams(Array.from(searchParams.entries()));
    if (query) {
      params.set("search", query);
    } else {
      params.delete("search");
    }
    router.replace(`?${params.toString()}`, { scroll: false });
  };

  const handleTabChange = (value: string) => {
    const params = new URLSearchParams(Array.from(searchParams.entries()));
    params.set("tab", value);
    router.replace(`?${params.toString()}`, { scroll: false });
  };

  const handleCatalogError = useCallback((error: Error) => {
    console.error("Catalog fetch failed:", error);
  }, []);

  // Catalog operation - unified data fetching
  const catalogOperation = useAsyncOperation<CatalogSchema>(fetchCatalog, {
    onError: handleCatalogError,
    showErrorToast: false,
  });

  // Load catalog data
  useEffect(() => {
    catalogOperation.execute();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [catalogOperation.execute]);

  return (
    <ErrorBoundary
      onError={(error: Error, errorInfo: ErrorInfo) => {
        console.error("Application error:", error, errorInfo);
        toast({
          description: "An unexpected error occurred. The page will reload automatically.",
          title: "Application Error",
          variant: "destructive",
        });

        // Auto-reload after 3 seconds
        setTimeout(() => {
          window.location.reload();
        }, 3000);
      }}
    >
      <Tabs
        className="space-y-6"
        onValueChange={handleTabChange}
        value={searchParams.get("tab") || "tiles"}
      >
        <TabsList className="grid w-full grid-cols-4">
          <TabsTrigger value="tiles">Data Catalog</TabsTrigger>
          <TabsTrigger value="styles">Styles Catalog</TabsTrigger>
          <TabsTrigger value="fonts">Font Catalog</TabsTrigger>
          <TabsTrigger value="sprites">Sprite Catalog</TabsTrigger>
        </TabsList>

        <TabsContent value="tiles">
          <TilesCatalog
            error={catalogOperation.error}
            isLoading={catalogOperation.isLoading}
            onSearchChangeAction={setSearchQuery}
            searchQuery={searchQuery}
            tileSources={catalogOperation.data?.tiles}
          />
        </TabsContent>

        <TabsContent value="styles">
          <StylesCatalog
            error={catalogOperation.error}
            isLoading={catalogOperation.isLoading}
            onSearchChangeAction={setSearchQuery}
            searchQuery={searchQuery}
            styles={catalogOperation.data?.styles}
          />
        </TabsContent>

        <TabsContent value="fonts">
          <FontCatalog
            error={catalogOperation.error}
            fonts={catalogOperation.data?.fonts}
            isLoading={catalogOperation.isLoading}
            onSearchChangeAction={setSearchQuery}
            searchQuery={searchQuery}
          />
        </TabsContent>

        <TabsContent value="sprites">
          <SpriteCatalog
            error={catalogOperation.error}
            isLoading={catalogOperation.isLoading}
            onSearchChangeAction={setSearchQuery}
            searchQuery={searchQuery}
            spriteCollections={catalogOperation.data?.sprites}
          />
        </TabsContent>
      </Tabs>

      <Toaster />
    </ErrorBoundary>
  );
}
