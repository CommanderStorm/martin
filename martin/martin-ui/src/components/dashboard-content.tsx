import { type ErrorInfo, useCallback, useEffect, useState } from "react";
import { FontCatalog } from "@/components/catalogs/font";
import { SpriteCatalog } from "@/components/catalogs/sprite";
import { StylesCatalog } from "@/components/catalogs/styles";
import { TilesCatalog } from "@/components/catalogs/tiles";
import { ErrorBoundary } from "@/components/error/error-boundary";
import { Tabs, TabsContent, TabsList, TabsTrigger } from "@/components/ui/tabs";
import { Toaster } from "@/components/ui/toaster";
import { useAsyncOperation } from "@/hooks/use-async-operation";
import { useToast } from "@/hooks/use-toast";
import { buildMartinUrl } from "@/lib/api";
import type { CatalogSchema } from "@/lib/types";

const fetchCatalog = async (): Promise<CatalogSchema> => {
  const res = await fetch(buildMartinUrl("/catalog"));
  if (!res.ok) {
    throw new Error(`Failed to fetch catalog: ${res.statusText}`);
  }
  return res.json();
};

export function DashboardContent() {
  const { toast } = useToast();
  const [activeTab, setActiveTab] = useState("tiles");
  const [searchQuery, setSearchQuery] = useState("");

  const handleTabChange = (value: string) => {
    setActiveTab(value);
  };

  const handleCatalogError = useCallback((error: Error) => {
    console.error("Catalog fetch failed:", error);
  }, []);

  // Catalog operation - unified data fetching
  const catalogOperation = useAsyncOperation<CatalogSchema>(fetchCatalog, {
    onError: handleCatalogError,
    showErrorToast: false,
  });

  // Load catalog data
  useEffect(() => {
    catalogOperation.execute();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [catalogOperation.execute]);

  return (
    <ErrorBoundary
      onError={(error: Error, errorInfo: ErrorInfo) => {
        console.error("Application error:", error, errorInfo);
        toast({
          description: "An unexpected error occurred. The page will reload automatically.",
          title: "Application Error",
          variant: "destructive",
        });

        // Auto-reload after 3 seconds
        setTimeout(() => {
          window.location.reload();
        }, 3000);
      }}
    >
      <Tabs className="space-y-6" onValueChange={handleTabChange} value={activeTab}>
        <TabsList className="grid w-full grid-cols-4">
          <TabsTrigger value="tiles">Data Catalog</TabsTrigger>
          <TabsTrigger value="styles">Styles Catalog</TabsTrigger>
          <TabsTrigger value="fonts">Font Catalog</TabsTrigger>
          <TabsTrigger value="sprites">Sprite Catalog</TabsTrigger>
        </TabsList>

        <TabsContent value="tiles">
          <TilesCatalog
            error={catalogOperation.error}
            isLoading={catalogOperation.isLoading}
            onSearchChangeAction={setSearchQuery}
            searchQuery={searchQuery}
            tileSources={catalogOperation.data?.tiles}
          />
        </TabsContent>

        <TabsContent value="styles">
          <StylesCatalog
            error={catalogOperation.error}
            isLoading={catalogOperation.isLoading}
            onSearchChangeAction={setSearchQuery}
            searchQuery={searchQuery}
            styles={catalogOperation.data?.styles}
          />
        </TabsContent>

        <TabsContent value="fonts">
          <FontCatalog
            error={catalogOperation.error}
            fonts={catalogOperation.data?.fonts}
            isLoading={catalogOperation.isLoading}
            onSearchChangeAction={setSearchQuery}
            searchQuery={searchQuery}
          />
        </TabsContent>

        <TabsContent value="sprites">
          <SpriteCatalog
            error={catalogOperation.error}
            isLoading={catalogOperation.isLoading}
            onSearchChangeAction={setSearchQuery}
            searchQuery={searchQuery}
            spriteCollections={catalogOperation.data?.sprites}
          />
        </TabsContent>
      </Tabs>

      <Toaster />
    </ErrorBoundary>
  );
}
