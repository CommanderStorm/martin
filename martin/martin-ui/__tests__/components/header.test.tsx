import { screen } from "@testing-library/react";
import React from "react";
import { Header } from "@/components/header";
import { render } from "../utils/test-utils";

jest.mock("@/components/theme-switcher", () => ({
  ThemeSwitcher: () => <div data-testid="theme-switcher">Theme Switcher Mock</div>
}));
jest.mock("next/image", () => ({
  __esModule: true,
  default: (props: any) => {
    // Convert boolean props to strings to avoid React DOM warnings
    const imgProps = {...props};
    if (typeof imgProps.priority === 'boolean') {
      imgProps.priority = imgProps.priority.toString();
    }
    return <img {...imgProps} />;
  }
}));

// Set environment variable used in Header
process.env.NEXT_PUBLIC_VERSION = 'v0.0.0-test';

describe("Header Component", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("renders correctly", () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });

  it("displays the Martin version", () => {
    const { getByText } = render(<Header />);
    expect(getByText("v0.0.0-test")).toBeInTheDocument();
  });

  it("contains navigation links", () => {
    const { getByText } = render(<Header />);

    const documentationLink = getByText("Documentation");
    expect(documentationLink).toBeInTheDocument();

    // hidden on mobile, so using a more specific query
    const aboutUsLink = getByText("About us");
    expect(aboutUsLink).toBeInTheDocument();
    expect(aboutUsLink.closest("a")).toHaveAttribute("href", "https://maplibre.org");
  });

  it("includes the theme switcher", () => {
    const { getByTestId } = render(<Header />);
    expect(getByTestId("theme-switcher")).toBeInTheDocument();
  });
});
