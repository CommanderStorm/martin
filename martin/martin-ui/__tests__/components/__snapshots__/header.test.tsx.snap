// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Header Component renders correctly 1`] = `
<div>
  <script
    nonce=""
  >
    ((e,s,n,l,o,d,u,h)=&gt;{let m=document.documentElement,w=["light","dark"];function p(r){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&d?o.map(f=&gt;d[f]||f):o;k?(m.classList.remove(...S),m.classList.add(d&&d[r]?d[r]:r)):m.setAttribute(y,r)}),C(r)}function C(r){h&&w.includes(r)&&(m.style.colorScheme=r)}function a(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(l)p(l);else try{let r=localStorage.getItem(s)||n,y=u&&r==="system"?a():r;p(y)}catch(r){}})("class","theme","light",null,["light","dark"],null,true,true)
  </script>
  <header
    class="border-b bg-navbar backdrop-blur-md backdrop-brightness-90 sticky top-0 z-50"
  >
    <div
      class="container mx-auto px-6"
    >
      <div
        class="flex h-20 items-center justify-between"
      >
        <div
          class="flex items-center space-x-4"
        >
          <div
            class="items-center flex"
          >
            <a
              class="hidden lg:flex"
              href="/"
            >
              <img
                alt="Logo of the Martin Tileserver"
                height="70"
                priority="true"
                src="/icon.png"
                title="Martin Tile Server"
                width="290"
              />
            </a>
            <h1
              class="text-3xl font-bold leading-relaxed md:block lg:hidden hidden"
            >
              MARTIN
            </h1>
          </div>
          <a
            class="items-center justify-center rounded-md border px-2 py-0.5 text-xs font-medium w-fit whitespace-nowrap shrink-0 [&>svg]:size-3 gap-1 [&>svg]:pointer-events-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive transition-[color,box-shadow] overflow-hidden border-transparent bg-primary text-primary-foreground [a&]:hover:bg-primary/90 hover:bg-purple-700 hidden md:block p-1"
            data-slot="badge"
            href="https://github.com/maplibre/martin/releases/tag/v0.0.0-test"
          >
            v0.0.0-test
          </a>
        </div>
        <div
          class="flex items-center space-x-6"
        >
          <a
            class="flex items-center gap-2 text-foreground hover:bg-accent hover:text-accent-foreground px-3 py-2 rounded-md transition-all"
            data-state="closed"
            href="https://maplibre.org/martin/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-book-open"
              fill="none"
              height="18"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 7v14"
              />
              <path
                d="M3 18a1 1 0 0 1-1-1V4a1 1 0 0 1 1-1h5a4 4 0 0 1 4 4 4 4 0 0 1 4-4h5a1 1 0 0 1 1 1v13a1 1 0 0 1-1 1h-6a3 3 0 0 0-3 3 3 3 0 0 0-3-3z"
              />
            </svg>
            <span>
              Documentation
            </span>
          </a>
          <a
            class="md:flex hidden items-center gap-2 text-foreground hover:bg-accent hover:text-accent-foreground px-3 py-2 rounded-md transition-all"
            data-state="closed"
            href="https://maplibre.org"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-info"
              fill="none"
              height="18"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="10"
              />
              <path
                d="M12 16v-4"
              />
              <path
                d="M12 8h.01"
              />
            </svg>
            <span>
              About us
            </span>
          </a>
          <div
            data-testid="theme-switcher"
          >
            Theme Switcher Mock
          </div>
        </div>
      </div>
    </div>
  </header>
</div>
`;
